clippy_lints/src/collapsible_if.rs

```diff
@@ -198,6 +204,12 @@ fn block_starts_with_comment(cx: &LateContext<'_>, block: &Block<'_>) -> bool {
     trimmed_block_text.starts_with("//") || trimmed_block_text.starts_with("/*")
 }
 
+fn span_not_contain_stmt(cx: &LateContext<'_>, span: Span) -> bool {
+    matches!(snippet_opt(cx, span), Some(text) if tokenize_with_text(&text[1..]).all(|(token, _, _)| {
+        matches!(token, TokenKind::LineComment { .. } | TokenKind::BlockComment { .. } | TokenKind::Whitespace)
+    }))
+}
+
```

Comment:
Could you please:
- use `span.get_source_text(cx)` to avoid allocating a string with the span content instead of `snippet_opt(cx, span)`
- move this function to `clippy_utils` (after `span_contains_comment`) with the following signature: `pub fn span_contains_non_comment(cx: &impl source::HasSession, span: Span) -> bool`
- add documentation